### PR TITLE
Add grpc backend to element compiler

### PR DESCRIPTION
**Pull request by NikolaBo**
This PR adds a go-grpc interceptor backend to the element compiler. Opening draft PR for element compiler before making changes to graph compiler. Issue: #3

### CODE REVIEW

File: compiler/README.md [hunk 1]
```diff
@@ -62,6 +62,10 @@ options:
                         Method Name (must be defined in proto)
   -b BACKEND, --backend BACKEND
                         Backend Code Target
+  -n MOD_NAME, --mod_name MOD_NAME
```
Comment (Romero027) [inline compiler/README.md:65]: I assume these two options are only for gRPC backend right? Can you add an explanation here?

File: compiler/element/__init__.py [hunk 2]
```diff
@@ -59,7 +64,7 @@ def gen_code(
     proto = os.path.basename(proto_path).replace(".proto", "")
 
     # Currently, we only support mRPC and Envoy (Proxy WASM) as the backends
```
Comment (Romero027) [inline compiler/element/__init__.py:66]: Can you update this comment?

File: compiler/element/backend/grpc/analyzer.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        name not in ctx.access_ops[ctx.current_procedure]
+        or ctx.access_ops[ctx.current_procedure][name] != MethodType.SET
+    ):
+        ctx.access_ops[ctx.current_procedure][name] = method
+
+
+class AccessAnalyzer(
+    Visitor
+):  # AccessAnalyzer is used to analyze the IR and record the access operations.
+    # The operations are then used to avoid unnecessary locks and RPC decode when generating the corresponding WASM code.
```
Comment (Romero027) [inline compiler/element/backend/grpc/analyzer.py:22]: nit: change `WASM` to `Go`

**Review by Romero027 — APPROVED** [review]
Looks good to me! Thanks for putting this together! 

Just a few nits and feel free to merge it.

btw, can you create an issue to track the TODOs for grpc? (e.g., state)

File: compiler/README.md (hunk 1 above)
Comment (NikolaBo) [reply compiler/README.md:65]: The supported backends section of this readme explains them, though I could move it up.

**Pull request merged**